lang/cs/Org.Apache.REEF.IO/FileSystem/Hadoop/HDFSCommandRunner.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using Org.Apache.REEF.IO.FileSystem.Hadoop.Parameters;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.IO.FileSystem.Hadoop
+{
+    /// <summary>
+    /// Helper class to execute commands
+    /// </summary>
+    internal sealed class HdfsCommandRunner
+    {
+        /// <summary>
+        /// The name of the <code>hdfs</code> command.
+        /// </summary>
+        private const string HdfsCommandName = "hdfs.cmd";
+
+        /// <summary>
+        /// The folder within <code>HadoopHome</code> that contains the <code>hdfs</code> command.
+        /// </summary>
+        private const string BinFolderName = "bin";
+
+        private static readonly Logger Logger = Logger.GetLogger(typeof(HdfsCommandRunner));
+
+        /// <summary>
+        /// Path to hdfs.cmd
+        /// </summary>
+        private readonly string _hdfsCommandPath;
+
+        /// <summary>
+        /// The number of retries on HDFS commands.
+        /// </summary>
+        private readonly int _numberOfRetries;
+
+        /// <summary>
+        /// The timeout on each of the retries.
+        /// </summary>
+        private readonly int _timeOutInMilliSeconds;
+
+        [Inject]
+        private HdfsCommandRunner([Parameter(typeof(NumberOfRetries))] int numberOfRetries,
+            [Parameter(typeof(CommandTimeOut))] int timeOutInMilliSeconds,
+            [Parameter(typeof(HadoopHome))] string hadoopHome)
+        {
+            _numberOfRetries = numberOfRetries;
+            _timeOutInMilliSeconds = timeOutInMilliSeconds;
+
+            if (hadoopHome == HadoopHome.DefaultValue)
```

Comment:
== works here if the client doesn't set it. What if the user set the that is the same a default value? Then you might want to use Equals.